aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -697,31 +670,43 @@ return AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
             other.const_data_ptr<scalar_t>(),
             incy,
             result.mutable_data_ptr<scalar_t>());
-
-        return result;
       });
 }
 
-Tensor vdot_cuda(const Tensor& self, const Tensor& other) {
+TORCH_IMPL_FUNC(dot_out_cuda)(const Tensor& self, const Tensor& other, const Tensor& result) {
+  dot_out_cuda_impl(self, other, result);
+}
+
+void vdot_out_cuda_impl(const Tensor& self, const Tensor& other, const Tensor& result) {
   if (!self.is_complex()) {
-    return dot_cuda(self, other);
+    dot_out_cuda_impl(self, other, result);
+    return;
   }
 
   if (self.is_conj()) {
     if (other.is_conj()) {
-      return vdot_cuda(other.conj(), self.conj());
+      vdot_out_cuda_impl(other.conj(), self.conj(), result);
     } else {
-      return dot_cuda(self.conj(), other);
+      dot_out_cuda_impl(self.conj(), other, result);
     }
+    return;
   } else if (other.is_conj()) {
-    return (dot_cuda(self, other.conj())).conj();
+    Tensor temp_result = at::empty({}, self.options());
+    dot_out_cuda_impl(self, other.conj(), temp_result);
+    result.fill_(temp_result.conj());
+    return;
   }
 
-  at::NoNamesGuard guard;
-  dot_check(self, other);
+  // Most checks are performed in the META_FUNC, this is CUDA specific
+  TORCH_CHECK(
+    (self.numel() <= INT_MAX) && (self.stride(0) <= INT_MAX) &&
+        (other.stride(0) <= INT_MAX),
+    "vdot only supports n, incx, incy with the bound [val] <= %d",
+    INT_MAX);
 
   if (self._is_zerotensor() || other._is_zerotensor()) {
-    return at::_efficientzerotensor({}, self.options());
+    result.fill_(0);
```

Comment:
The behavior is changed - before, this path would use ZeroTensor allocator: https://github.com/pytorch/pytorch/blob/ff616c26fb66b26966b6f6f50aa85bf615400816/aten/src/ATen/native/TensorFactories.h#L119-L120, which seems to not fill result with 0.